File: Easy/0217_contains-duplicate/contains-duplicate.go
package main

import (
	"fmt"
	"sort"
)

func containsDuplicate(nums []int) bool {
	sort.Ints(nums)
	for i := 0; i < len(nums) - 1; i++ {
		if nums[i] == nums[i+1] {
			return true
		}
	}
	return false
}

func main() {
	tests := [][]int{
		{1,2,3,1},
		{1,2,3,4},
		{1,1,1,3,3,4,3,2,4,2},
	}

	for _, nums := range tests {
		fmt.Printf("---\nnums: %v\n", nums)
		fmt.Printf("containsDuplicate: %t\n", containsDuplicate(nums))
	}
}

File: Easy/0217_contains-duplicate/contains-duplicate.py
#!/bin/python3

from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set(nums)
        if len(nums) == len(seen):
            return False
        return True

def main():
    tests: List[List[int]] = [
        [1,2,3,1],
        [1,2,3,4],
        [1,1,1,3,3,4,3,2,4,2],
    ]

    for nums in tests:
        print(f"---\nnums= {nums}")
        print(f"containsDuplicate: {Solution().containsDuplicate(nums)}")

if __name__ == "__main__":
    main()
